From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Mon, 18 Sep 2017 20:22:48 +0200
Subject: [PATCH] {BUG in 4.14-rc1} mm/hmm: initialize entry before using it

clang reports:

    mm/hmm.c:423:24: error: variable 'entry' is uninitialized when used
    here [-Werror,-Wuninitialized]
                            if (!non_swap_entry(entry)) {
                                                ^~~~~
    mm/hmm.c:421:4: note: variable 'entry' is declared here
                            swp_entry_t entry;
                            ^
---
 mm/hmm.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mm/hmm.c b/mm/hmm.c
index a88a847bccba..831892f0d749 100644
--- a/mm/hmm.c
+++ b/mm/hmm.c
@@ -420,14 +420,14 @@ static int hmm_vma_walk_pmd(pmd_t *pmdp,
 		if (!pte_present(pte)) {
 			swp_entry_t entry;
 
+			entry = pte_to_swp_entry(pte);
+
 			if (!non_swap_entry(entry)) {
 				if (hmm_vma_walk->fault)
 					goto fault;
 				continue;
 			}
 
-			entry = pte_to_swp_entry(pte);
-
 			/*
 			 * This is a special swap entry, ignore migration, use
 			 * device and report anything else as error.
-- 
